docs/source/class_basics.rst

```diff
@@ -194,50 +194,96 @@ effect at runtime:
 
 .. code-block:: python
 
-   class A:
+   class Base:
        def inc(self, x: int) -> int:
            return x + 1
 
-   class B(A):
+   class Derived(Base):
        def inc(self, x):   # Override, dynamically typed
-           return 'hello'  # Incompatible with 'A', but no mypy error
+           return 'hello'  # Incompatible with 'Base', but no mypy error
 
 Abstract base classes and multiple inheritance
 **********************************************
 
 Mypy supports Python abstract base classes (ABCs). Abstract classes
 have at least one abstract method or property that must be implemented
-by a subclass. You can define abstract base classes using the
-``abc.ABCMeta`` metaclass, and the ``abc.abstractmethod`` and
-``abc.abstractproperty`` function decorators. Example:
+by any *concrete* (non-abstract) subclass. You can define abstract base
+classes using the ``abc.ABCMeta`` metaclass, and the ``abc.abstractmethod``
+and ``abc.abstractproperty`` function decorators. Example:
```

Comment:
Actually `@abstractproperty` has been deprecated since 3.3 -- just put`@property` above `@abstractmethod`: https://docs.python.org/3/library/abc.html?highlight=abstractproperty#abc.abstractproperty . (However, in 2.7 you still need `@abstractproperty`.)